go/ngql/README.md

```diff
@@ -0,0 +1,39 @@
+# Noms GraphQL
+
+An experimental bridge between noms and [GraphQL](http://graphql.org/)
+
+# Status
+
+ * All Noms types are supported except
```

Comment:
It's kind of a grab bag. I'm not aware of any fundamental issues.

Just unimplemented:
-Type (a few ways to do this: use graphql types or define our own)
-Blob (presumably, we'd just define Blob scalar type and just use base64)

graphql-go impl issue:
-Nested collections. JS appears to allow this. Option 1) add support, Option 2) just get more verbose and require all collections to have a containing struct. can kind of see whether its in our way.

unsure:
-Unions with non-struct component types. Don't see any reason why this wouldn't work, but again go impl doesn't support

Mutations:
-multiple ways to do this. my idea right now is to basically support a single `update` mutation which takes a collection of arguments which sort of logically are the changes we'd encode in the graph-mutator.

Set intersection:
-requires doing some refactoring in noms so that datas can use paths (right now creates cyclic package references).

Perf: I suspect there would be decent gains (in relative terms) to be had optimizing the graphql-go and our chunk fetching strategy, but my guess is that perf won't be in our way. the whole idea here is to fetch relatively small amounts of data.